crates/matrix-sdk-ui/src/timeline/event_item/content.rs

```diff
@@ -402,6 +416,27 @@ impl ReactionGroup {
     pub fn senders(&self) -> impl Iterator<Item = &UserId> {
         self.values().map(AsRef::as_ref)
     }
+
+    /// All reactions within this reaction group that were sent by the given
+    /// user Note that it is possible for a user to have sent multiple
+    /// reactions with the same key
+    pub fn by_sender(
+        &self,
+        user_id: &UserId,
+    ) -> impl Iterator<Item = (Option<&OwnedTransactionId>, Option<&OwnedEventId>)> {
+        let user_id = user_id.to_owned();
+        self.as_refs().filter_map(move |(k, v)| match v == &user_id {
+            true => Some(k),
+            false => None,
+        })
+    }
```

Comment:
```suggestion
    pub fn by_sender<'a>(
        &'a self,
        user_id: &'a UserId,
    ) -> impl Iterator<Item = (Option<&OwnedTransactionId>, Option<&OwnedEventId>)> + 'a {
        self.as_refs().filter(|(_, v)| v == user_id).map(|(k, _) k)
    }
```

Reply:
That surprises me a bit, but the `move` wasn't really what my suggestion was about anyways :)